dpnp/backend/kernels/dpnp_krnl_mathematical.cpp

```diff
@@ -277,6 +277,43 @@ void dpnp_remainder_c(void* array1_in, void* array2_in, void* result1, size_t si
     event.wait();
 }
 
+template <typename _KernelNameSpecialization1, typename _KernelNameSpecialization2, typename _KernelNameSpecialization3>
+class dpnp_trapz_c_kernel;
+
+template <typename _DataType_input1, typename _DataType_input2, typename _DataType_output>
+void dpnp_trapz_c(void* array1_in, void* array2_in, void* result1, double dx, size_t array1_size, size_t array2_size)
+{
+    _DataType_input1* array1 = reinterpret_cast<_DataType_input1*>(array1_in);
+    _DataType_input2* array2 = reinterpret_cast<_DataType_input2*>(array2_in);
+    _DataType_output* result = reinterpret_cast<_DataType_output*>(result1);
+
+    if (array2_size < 2) {
+        _DataType_input1* sum = reinterpret_cast<_DataType_input1*>(dpnp_memory_alloc_c(1 * sizeof(_DataType_input1)));
+
+        dpnp_sum_c<_DataType_input1>(array1, sum, array1_size);
+
+        result[0] = sum[0];
+
+        dpnp_memory_free_c(sum);
```

Comment:
Why not?

Reply:
because just this is enough, if 1 elem. Just math :)
```cpp
        result[0] = 0;
```

Reply:
`result` and `sum` could be of different types. So `result[0] = sum[0];` provides implicit casting.

Reply:
> because just this is enough, if 1 elem. Just math :)

at least for cases when `array1_size` == `array2_size`, that implemented here.

> `result` and `sum` could be of different types. So `result[0] = sum[0];` provides implicit casting.

@densmirn no implicit casting for `result[0] = 0;`

Reply:
Why do you want to keep `result[0] = 0` if we need to compute sum of `array1` elements?

Reply:
BTW `array1` is not one-element array.

Reply:
Due to trapezoidal rule, you can see https://en.wikipedia.org/wiki/Trapezoidal_rule .

Reply:
Again `array1` is not one-element array.

Reply:
> BTW array1 is not one-element array.

?

```cpp
if (array2_size < 2)
{
    result[0] = 0;
} // array1_size == array2_size 
```

Reply:
`array1_size == array2_size` is not always True.

Reply:
@densmirn

1)  this backend func implemented for case `array1_size == array2_size`. The check provided on python level.
Also in this case, there is no need for two different size values (array1_size , array2_size) containing the same value.
2) This func is not implemented well if `array1_size` ,` array2_size` are different.

Reply:
> Also in this case, there is no need for two different size values (array1_size , array2_size) containing the same value.

But I think we can save this params for API.